static/src/javascripts/projects/common/modules/discussion/loader.js

```diff
@@ -0,0 +1,580 @@
+                container.html(newPagination);
+            }
+        });
+
+        this.setState('loading');
+
+        this.on('user:loaded', () => {
+            this.initState();
+            this.renderCommentBar();
+
+            if (this.user) {
+                this.comments.addUser(this.user);
+
+                const userPageSize = userPrefs.get('discussion.pagesize');
+                let pageSize = defaultPagesize;
+
+                if (typeof userPageSize === 'number') {
+                    pageSize = userPageSize;
+                } else if (userPageSize === 'All') {
+                    pageSize = config.switches.discussionAllPageSize
+                        ? 'All'
+                        : 100;
+                }
+
+                this.initPageSizeDropdown(pageSize);
+
+                if (
+                    config.switches.discussionPageSize &&
+                    isBreakpoint({ min: 'tablet' })
+                ) {
+                    this.comments.options.pagesize = pageSize;
+                }
+
+                if (this.user.isStaff) {
+                    this.removeState('not-staff');
+                    this.setState('is-staff');
+                }
+            }
+
+            // Only truncate the loaded comments on this initial fetch,
+            // and when no comment ID or #comments location is present.
+            const shouldTruncate =
+                !commentId && window.location.hash !== '#comments';
+
+            this.loadComments({
+                comment: commentId,
+                shouldTruncate,
+            }).catch(() => this.logError('Comments'));
+        });
+
+        this.getUser();
+    }
+
+    initTopComments(): Promise<void> {
+        const discussionId = this.getDiscussionId();
+
+        this.on('click', '.js-jump-to-comment', (e: Event) => {
+            e.preventDefault();
+            const commentId = bonzo(e.currentTarget).data('comment-id');
+            this.gotoComment(commentId);
+        });
+
+        if (!discussionId) {
+            return Promise.resolve();
+        }
+
+        return fetchJson(
+            `/discussion/top-comments/${discussionId}.json?commentsClosed=${this.getDiscussionClosed().toString()}`,
+            {
+                mode: 'cors',
+            }
+        )
+            .then(resp => {
+                this.$topCommentsContainer.html(resp.html);
+                this.topCommentCount = qwery(
+                    '.d-top-comment',
+                    this.$topCommentsContainer[0]
+                ).length;
+                if (this.topCommentCount !== 0) {
+                    $('.js-discussion-comment-box--bottom').removeClass(
+                        'discussion__comment-box--bottom--hidden'
+                    );
+                    this.setState('has-top-comments');
+                }
+            })
+            .catch(() => this.logError('Top comments'));
+    }
+
+    logError(commentType: string, error?: Error) {
```

Comment:
Yeah, I simplified the logic, because during the typing I started to have the feeling that most of the conditions could never be fullfilled. I had a long into sentry, which confirmed that.